File: python/백준/부르트 포스/1018 체스판 다시 칠하기.py
n,m = map(int, input().split())

board = []
result = []

for _ in range(n):
  board.append(input())

for i in range(n - 7):
  for j in range(m - 7):
    draw_white = 0
    draw_black = 0
    
    for a in range(i, i + 8):
      for b in range(j, j + 8):
          if (a + b) % 2 == 0:
            if board[a][b] != "W":
              draw_white += 1
            else:
              draw_black += 1
          else:
            if board[a][b] != "W":
              draw_black += 1
            else: 
              draw_white += 1
    
    result.append(min(draw_black, draw_white))
          
print(min(result))

File: cpp/백준/부르트 포스/1018 체스판 다시 칠하기.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <climits>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, m;
    cin >> n >> m;

    vector<string> board(n);
    for (int i = 0; i < n; ++i) cin >> board[i];

    int best = INT_MAX;

    for (int i = 0; i <= n - 8; ++i) {
        for (int j = 0; j <= m - 8; ++j) {
            int draw_white = 0;
            int draw_black = 0;

            for (int a = i; a < i + 8; ++a) {
                for (int b = j; b < j + 8; ++b) {
                    if ((a + b) % 2 == 0) {
                        if (board[a][b] != 'W') draw_white++;
                        else draw_black++;
                    } else {
                        if (board[a][b] != 'W') draw_black++;
                        else draw_white++;
                    }
                }
            }

            best = min(best, min(draw_black, draw_white));
        }
    }

    cout << best << "\n";
    return 0;
}
